### PR TITLE
feat: file sync addon

### LINKED ISSUE #75: Sync command

### PR DESCRIPTION
**Pull request by Xstoudi**
Fixes: #75

### CODE REVIEW

File: addons/file-sync/Commands/Sync.ts [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+    await file.save();
+  }
+
+  private wait() {
+    return new Promise((resolve) => {
+      this.logger.info(`waiting ${this.interval}s...`);
+      setTimeout(resolve, this.interval * 1000);
+    });
+  }
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: you can use `setTimeout` from `timers/promises`

File: addons/file-sync/Commands/Sync.ts [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      );
+      return;
+    }
+
+    this.logger.debug('stats have changed, proceed', importConfigId, filename);
+
+    if (lastRevision.status === 'PENDING') {
+      this.logger.debug(
+        'latest revision is still pending, update',
+        undefined,
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: ?

File: addons/file-sync/Commands/Sync.ts [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        filename,
+      );
+
+      file.revisions[0] = {
+        ...lastRevision,
+        size,
+        creationDate,
+        modificationDate,
+      };
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: `Object.assign(lastRevision, { fieldsYouChange })`

File: addons/file-sync/Commands/Sync.ts (hunk 2 above)
Comment (Xstoudi) [reply addons/file-sync/Commands/Sync.ts]: image

File: addons/file-sync/Commands/Sync.ts [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      importsConfigToProcess.push(...importConfigs);
+    }
+
+    const syncCount = (
+      await Promise.all(
+        importsConfigToProcess.map((importConfig) =>
+          this.executeConfig(importConfig),
+        ),
+      )
+    ).reduce((a, b) => a + b, 0);
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: we decided to do it in sequence, not parallel

File: addons/file-sync/Commands/Sync.ts (hunk 2 above)
Comment (targos) [reply addons/file-sync/Commands/Sync.ts]: my question is what's the point of this `undefined` in the log?

File: addons/file-sync/Commands/Sync.ts (hunk 4 above)
Comment (Xstoudi) [reply addons/file-sync/Commands/Sync.ts]: is it better ?

File: addons/file-sync/Commands/Sync.ts [hunk 5]
```diff
@@ -63,23 +63,17 @@ export default class Sync extends BaseCommand {
...
-    const syncCount = (
-      await Promise.all(
-        importsConfigToProcess.map((importConfig) =>
-          this.executeConfig(importConfig),
-        ),
-      )
-    ).reduce((a, b) => a + b, 0);
-
-    this.logger.success(`${syncCount} files synchronized`);
+    for await (const importConfigToProcess of importConfigsToProcess) {
```
Comment (targos) [inline addons/file-sync/Commands/Sync.ts]: I don't think you need the await here (it's not an array of promises)

Comment (targos) [reply addons/file-sync/Commands/Sync.ts]: oops

Comment (Xstoudi) [reply addons/file-sync/Commands/Sync.ts]: Yeah, I insta-fixed it ^^

**Pull request merged**